### DISCUSSION TITLE
MQTT feedback when I physically press the tasmota button

### DISCUSSION THREAD

**Discussion by flayy73** [arendst/Tasmota#21398] [Support request]
I would like to have an MQTT feedback (stat/mydevice/power = ON/OFF) when I physically press the switch.
Is it possible?

**Comment by sfromis** [comment]
You do already get that MQTT feedback when power goes on/off. If you **instead** want the button action, you could decouple it from the relay action (producing messages), and then have rules to switch the relay if that's what you want.

**Comment by flayy73** [comment]
Tks for the feedback. The switch is set as switchmode 3 (is a button). When I press the button, I don't see any change on the relevant MQTT topic even with buttontopic set to 1.
If I act the switch from the web UI or from HA, I get MQTT output but when I physically push the button, I don't have any change on the MQTT topic.

**Reply by barbudor** [reply]
If it's a switch, then buttontopic has no effect
Read about switch topic, or better switchmode 15

**Reply by barbudor** [reply]
Actually you are not clear on what you have and want
POWER ON/OFF status are sent by relay (or similar) output when they change state, not by buttons nor switches

Please provide mode details on what you have as Relay, lights, buttons and switches and explain better how you want that to behave and interact with MQTT

**Comment by flayy73** [comment]
Ok, tasmota is loaded into a Shelly1 device. 
Module parameter is Shelly1 (46).
Switchmode is set to 1 (ON when the switch is pressed and OFF when is released, right?).
I would like to know, by an MQTT topic,  when I manually push the switch.
Thanks

**Reply by sfromis** [reply]
https://tasmota.github.io/docs/Buttons-and-Switches/#detach-switches-with

But still, you **already** get MQTT messages when the Power channel is toggled by the input action.
